src/Sparrow/Platform/Posix/PosixSodium.cs

```diff
@@ -32,90 +38,95 @@ static PosixSodium()
             }
             else
             {
-                X64.Initialize();
+                if (PlatformDetails.RunningOnMacOsx)
+                    MacOsxX64.Initialize();
+                else
+                    X64.Initialize();
+            }
+
+            // test for existance of libsodium:
+            try
+            {
+                crypto_sign_publickeybytes();
+            }
+            catch (Exception e)
+            {
+                if (_isMac64)
+                    throw new Exception("Make sure libsodium is installed on your Mac Osx. Use `brew install libsodium`", e);
+
+                if (PlatformDetails.RunningOnPosix)
+                        throw new Exception("Make sure libsodium is installed on your Linux OS. (install package `libsodium` or `libsodium-18`", e);
+
+                throw new Exception("Make sure libsodium is installed on your Windows OS.", e);
             }
         }
 
         // ReSharper disable once InconsistentNaming
         public static UIntPtr crypto_generichash_bytes()
         {
-            return _is32bits ?
-                (_isArm ? Arm.crypto_generichash_bytes() : X86.crypto_generichash_bytes()) :
-                X64.crypto_generichash_bytes();
+            return _is32bits ? (_isArm ? Arm.crypto_generichash_bytes() : X86.crypto_generichash_bytes()) : _isMac64 ? MacOsxX64.crypto_generichash_bytes() : X64.crypto_generichash_bytes();
```

Comment:
Cost is going to be dominated by hash and marshalling. The structure of that code is going to be very packed. There is hardly any improvement that can be made unless we are going to do some bond like stuff that would only make sense if we decide to tackle the most general issue.